framework/test/unit/modules/interoperability/sidechain/commands/message_recovery.spec.ts

```diff
@@ -0,0 +1,300 @@
+		terminatedOutboxSubstore = stateStore.getStore(
+			MODULE_ID_INTEROPERABILITY,
+			STORE_PREFIX_TERMINATED_OUTBOX,
+		);
+		mockGetStore = jest.fn();
+		when(mockGetStore)
+			.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_TERMINATED_OUTBOX)
+			.mockReturnValue(terminatedOutboxSubstore);
+		sidechainInteroperabilityStore = new SidechainInteroperabilityStore(
+			MODULE_ID_INTEROPERABILITY,
+			mockGetStore,
+			new Map(),
+		);
+		messageRecoveryCommand = new MessageRecoveryCommand(
+			MODULE_ID_INTEROPERABILITY,
+			interoperableCCAPIs,
+			ccCommands,
+		);
+
+		ccms = [
+			{
+				nonce: BigInt(0),
+				moduleID: 1,
+				crossChainCommandID: 1,
+				sendingChainID: 2,
+				receivingChainID: 3,
+				fee: BigInt(1),
+				status: 0,
+				params: Buffer.alloc(0),
+			},
+			{
+				nonce: BigInt(1),
+				moduleID: 1,
+				crossChainCommandID: 1,
+				sendingChainID: 4,
+				receivingChainID: 5,
+				fee: BigInt(2),
+				status: 0,
+				params: Buffer.alloc(0),
+			},
+		];
+		ccmsEncoded = ccms.map(ccm => codec.encode(ccmSchema, ccm));
+		merkleTree = new MerkleTree();
+		await merkleTree.init(ccmsEncoded);
+		queryHashes = [];
+		for (const data of ccmsEncoded) {
+			const leafValueWithoutNodeIndex = Buffer.concat(
+				[LEAF_PREFIX, data],
+				LEAF_PREFIX.length + data.length,
+			);
+			const leafHash = hash(leafValueWithoutNodeIndex);
+			queryHashes.push(leafHash);
+		}
+		generatedProof = await merkleTree.generateProof(queryHashes);
+		terminatedChainOutboxSize = generatedProof.size;
+		proof = {
+			size: terminatedChainOutboxSize,
+			indexes: generatedProof.idxs as number[],
+			siblingHashes: generatedProof.siblingHashes as Buffer[],
+		};
+		hashedCCMs = ccmsEncoded.map(ccm => hash(ccm));
+		outboxRoot = regularMerkleTree.calculateRootFromUpdateData(hashedCCMs, proof);
+		transactionParams = {
+			chainID: 3,
+			crossChainMessages: [...ccmsEncoded],
+			idxs: proof.indexes,
+			siblingHashes: proof.siblingHashes,
+		};
+		chainID = getIDAsKeyForStore(transactionParams.chainID);
+		encodedTransactionParams = codec.encode(messageRecoveryParamsSchema, transactionParams);
+
+		transaction = new Transaction({
+			moduleID: MODULE_ID_INTEROPERABILITY,
+			commandID: COMMAND_ID_MESSAGE_RECOVERY,
+			fee: BigInt(100000000),
+			nonce: BigInt(0),
+			params: encodedTransactionParams,
+			senderPublicKey: getRandomBytes(32),
+			signatures: [],
+		});
+		commandVerifyContext = createTransactionContext({
+			transaction,
+		}).createCommandVerifyContext<MessageRecoveryParams>(messageRecoveryParamsSchema);
+
+		jest
+			.spyOn(messageRecoveryCommand, 'getInteroperabilityStore' as any)
+			.mockImplementation(() => sidechainInteroperabilityStore);
+
+		await terminatedOutboxSubstore.setWithSchema(
+			chainID,
+			{
+				outboxRoot,
+				outboxSize: terminatedChainOutboxSize,
+				partnerChainInboxSize: 1,
+			},
+			terminatedOutboxSchema,
+		);
+	});
+
+	it('should return error if the sidechain outbox root is not valid', async () => {
```

Comment:
Tests also seems to be duplicated. If the code is the same, do we need 2 same tests?

Reply:
Unit tests for `verify` hook under `Sidechain MessageRecoveryCommand` removed with a note.